RadarSDK/RadarNotificationHelper.m

```diff
@@ -68,4 +74,149 @@ + (void)showNotificationsForEvents:(NSArray<RadarEvent *> *)events {
+    if (NSClassFromString(@"XCTestCase") == nil) {
+        [[UNUserNotificationCenter currentNotificationCenter] getPendingNotificationRequestsWithCompletionHandler:^(NSArray<UNNotificationRequest *> *_Nonnull requests) {
+            NSMutableSet *pendingIdentifiers = [NSMutableSet new];
+                
+            for (UNNotificationRequest *request in requests) {
+                [pendingIdentifiers addObject:request.identifier];
+            }
+                
+            for (UNNotificationRequest *request in registeredNotifications) {
+                if (![pendingIdentifiers containsObject:request.identifier]) {
+                    [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:[NSString stringWithFormat:@"Found pending notification | identifier = %@", request]];
+                    NSDate *lastCheckedTime = [RadarState lastCheckedOnPremiseNotification];
+                    [Radar logConversionWithNotification:request eventName:@"delivered_on_premise_notification" conversionSource:nil deliveredAfter:lastCheckedTime];
+                    // prevent double counting of the same notification
+                    [RadarState removePendingNotificationRequest:request];
+                }
+            }            
+        }];
+    }
+    [RadarState lastCheckedOnPremiseNotification];
+}
+
++ (void)removePendingNotificationsWithCompletionHandler:(void (^)(void))completionHandler {
+    UNUserNotificationCenter *notificationCenter = [UNUserNotificationCenter currentNotificationCenter];
+    [notificationCenter getPendingNotificationRequestsWithCompletionHandler:^(NSArray<UNNotificationRequest *> *_Nonnull requests) {
+        [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:[NSString stringWithFormat:@"Found %lu pending notifications", (unsigned long)requests.count]];
+        NSMutableArray *identifiers = [NSMutableArray new];
+        for (UNNotificationRequest *request in requests) {
+            if ([request.identifier hasPrefix:kSyncGeofenceIdentifierPrefix]) {
+                [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelInfo message:[NSString stringWithFormat:@"Found pending notification | identifier = %@", request.identifier]];
+                [identifiers addObject:request.identifier];
+            }
+        }
+
+        if (identifiers.count > 0) {
+            [notificationCenter removePendingNotificationRequestsWithIdentifiers:identifiers];
+            [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:@"Removed pending notifications"];
+        }
+
+        completionHandler();
+    }];
+}
+
++ (void)addOnPremiseNotificationRequests:(NSArray<UNNotificationRequest *> *)requests {
+    UNUserNotificationCenter *notificationCenter = [UNUserNotificationCenter currentNotificationCenter];
+    [notificationCenter getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings *settings) {
```

Comment:
Let's use this to grab the authorization status and stash it in state as well

Reply:
I agree that this its probably something we want to do and send up like we do send up location authorization.
However this is slightly off tangent to this PR and requires some server work as well, should we do this in another PR?